src/tests.c

```diff
@@ -5262,17 +5262,19 @@ void test_ecdsa_sign_verify(void) {
     secp256k1_scalar msg, key;
     secp256k1_scalar sigr, sigs;
     int getrec;
-    /* Initialize recid to suppress a false positive -Wconditional-uninitialized in clang.
-       VG_UNDEF ensures that valgrind will still treat the variable as uninitialized. */
-    int recid = -1; VG_UNDEF(&recid, sizeof(recid));
+    int recid;
     random_scalar_order_test(&msg);
     random_scalar_order_test(&key);
     secp256k1_ecmult_gen(&ctx->ecmult_gen_ctx, &pubj, &key);
     secp256k1_ge_set_gej(&pub, &pubj);
     getrec = secp256k1_testrand_bits(1);
-    random_sign(&sigr, &sigs, &key, &msg, getrec?&recid:NULL);
```

Comment:
fwiw, assigning the pointer to a variable also seems to solve this,
Isolated case: https://godbolt.org/z/q1T6eMccT

Reply:
I see, yeah.   I tried a variant of this, which didn't work. I think both "fixes" are fine, I'm going to keep mine if noone objects. 

Interestingly, your test case shows that the variant appears also on clang 12. But I can't get valgrind to complain on clang 12 output...